delphi-frontend/src/main/java/au/com/integradev/delphi/antlr/ast/node/TextLiteralNodeImpl.java

```diff
@@ -44,50 +50,130 @@ public <T> T accept(DelphiParserVisitor<T> visitor, T data) {
-        }
-      }
-      imageBuilder.append("'");
-      image = imageBuilder.toString();
+      image =
+          getChildren().stream()
+              .map(
+                  child -> {
+                    String result = child.getImage();
+                    if (child.getTokenType() == DelphiTokenType.ESCAPED_CHARACTER) {
+                      result = '^' + result;
+                    }
+                    return result;
+                  })
+              .collect(Collectors.joining());
     }
     return image;
   }
 
+  @Override
+  public String getValue() {
+    if (value == null) {
+      value = createValue();
+    }
+    return value;
+  }
+
+  @SuppressWarnings("removal")
   @Override
   public CharSequence getImageWithoutQuotes() {
-    return getStringWithoutQuotes(getImage());
+    return getValue();
+  }
+
+  private String createValue() {
+    if (isMultiline()) {
+      return createMultilineValue();
+    } else {
+      return createSingleLineValue();
+    }
+  }
+
+  private String createMultilineValue() {
+    Deque<String> lines =
+        getChild(0).getImage().lines().collect(Collectors.toCollection(ArrayDeque<String>::new));
+
+    lines.removeFirst();
+
+    String last = lines.removeLast();
+    String indentation = readLeadingWhitespace(last);
+
+    return lines.stream()
+        .map(line -> StringUtils.removeStart(line, indentation))
+        .collect(Collectors.joining("\n"));
+  }
+
+  private static String readLeadingWhitespace(String input) {
+    StringBuilder result = new StringBuilder();
+    for (int i = 0; i < input.length(); ++i) {
+      char c = input.charAt(i);
+      if (c <= 0x20 || c == 0x3000) {
+        result.append(c);
+      } else {
+        break;
+      }
+    }
+    return result.toString();
   }
 
-  private static CharSequence getStringWithoutQuotes(String string) {
-    return string.subSequence(1, string.length() - 1);
+  private String createSingleLineValue() {
+    StringBuilder imageBuilder = new StringBuilder();
+
+    for (DelphiNode child : getChildren()) {
+      switch (child.getTokenType()) {
+        case QUOTED_STRING:
+          String stringImage = child.getImage();
+          stringImage = stringImage.substring(1, stringImage.length() - 1);
+          stringImage = stringImage.replace("''", "'");
+          imageBuilder.append(stringImage);
+          break;
+
+        case CHARACTER_ESCAPE_CODE:
+          imageBuilder.append(characterEscapeToChar(child.getImage()));
+          break;
+
+        case ESCAPED_CHARACTER:
+          imageBuilder.append((char) ((child.getImage().charAt(0) + 64) % 128));
+          break;
+
+        default:
+          // Do nothing
+      }
+    }
+
+    return imageBuilder.toString();
+  }
+
+  private static char characterEscapeToChar(String image) {
+    image = image.substring(1);
+    int radix = 10;
+
+    switch (image.charAt(0)) {
+      case '$':
+        radix = 16;
+        image = image.substring(1);
```

Comment:
No need to assign image here - it's already assigned the same value above.

Reply:
This is taking off the leading `$`.